ui/src/features/get-jobs.js

```diff
@@ -43,12 +44,19 @@ class getJobs extends React.Component {
             sortedJobs: [],
             rerunJob: null,
             editJob: null,
+            openViewEditJob: false,
+            jobForEdit: null
         };
     }
 
     componentDidUpdate(prevProps) {
         if (prevProps.jobs !== this.props.jobs) {
             this.setState({ sortedJobs: [...this.props.jobs] });
+            const { match: { params, path } } = this.props;
+            if (path === '/jobs/:jobId/edit') {
+                const data = this.props.jobs.find((job) => job.id === params.jobId);
+                data && this.onEdit(data);
```

Comment:
first of all thank you for this Pr , it's awesome!
if data does not exist, think we should redirect to some not found page which we don't have right now
so maybe for now just make replace to the URL - by history.replace('/jobs'); to be consistence with the page state.

Reply:
added